### PR TITLE
DOCS-4446: document timestamped state stores

### PR DESCRIPTION
**Pull request by JimGalasyn**
Add a section on the timestamped state store interfaces, per [KIP-258](https://cwiki.apache.org/confluence/display/KAFKA/KIP-258%3A+Allow+to+Store+Record+Timestamps+in+RocksDB).

### CODE REVIEW

File: docs/streams/developer-guide/processor-api.html [hunk 1]
```diff
@@ -49,6 +49,7 @@
                         <li><a class="reference internal" href="#defining-and-creating-a-state-store" id="id4">Defining and creating a State Store</a></li>
                         <li><a class="reference internal" href="#fault-tolerant-state-stores" id="id5">Fault-tolerant State Stores</a></li>
                         <li><a class="reference internal" href="#enable-or-disable-fault-tolerance-of-state-stores-store-changelogs" id="id6">Enable or Disable Fault Tolerance of State Stores (Store Changelogs)</a></li>
+                        <li><a class="reference internal" href="#timestamped-state-stores" id="id11">Timestamped State Stores</a></li>
```
Comment (mjsax) [inline docs/streams/developer-guide/processor-api.html:52]: Should we also extend `Defining and creating a State Store` ? -- It seems the section you added below, focuses on their usage?

File: docs/streams/developer-guide/processor-api.html [hunk 2]
```diff
@@ -343,6 +344,38 @@ <h2>
 </pre></div>
                 </div>
             </div>
+            <div class="section" id="timestamped-state-stores">
+                <span id="streams-developer-guide-state-store-timestamps"></span><h3><a class="toc-backref" href="#id11">Timestamped State Stores</a><a class="headerlink" href="#timestamped-state-stores" title="Permalink to this headline"></a></h3>
+                <p>
+                    Starting in Kafka Streams 2.3, you can store record timestamps in KTables.
```
Comment (mjsax) [inline docs/streams/developer-guide/processor-api.html]: > Starting in Kafka Streams 2.3

Do we need this?

> can store

`KTables` always store timestamps by default. Should we rephrase?

File: docs/streams/developer-guide/processor-api.html [hunk 3]
```diff
@@ -343,6 +344,38 @@ <h2>
...
+                    Use <a class="reference external" href="/{{version}}/javadoc/org/apache/kafka/streams/state/TimestampedKeyValueStore.html">TimestampedKeyValueStore</a>
+                    when you need a key-(value/timestamp) store that supports put/get/delete and range queries. 
+                </p>
+                <p>
+                    Use <a class="reference external" href="/{{version}}/javadoc/org/apache/kafka/streams/state/TimestampedWindowStore.html">TimestampedWindowStore</a>
+                    when you need to store windowedKey-(value/timestamp) pairs.</p>
+                <p>
+                    The <a class="reference external" href="/{{version}}/javadoc/org/apache/kafka/streams/state/Stores.html">Stores</a>
+                    class provides corresponding factory methods for timestamped stores.</p>
+                <p>
```
Comment (mjsax) [inline docs/streams/developer-guide/processor-api.html]: Seems this part belong to section `Defining and creating a State Store` ?

File: docs/streams/developer-guide/processor-api.html [hunk 4]
```diff
@@ -343,6 +344,38 @@ <h2>
...
+                    <b>Upgrade note:</b> All users upgrade with a single rolling bounce per instance.
+                    <ul class="first simple">
+                        <li>For Platform API users, nothing changes in existing applications, and you
```
Comment (mjsax) [inline docs/streams/developer-guide/processor-api.html]: `Platform API` ? Do you mean `Processor API` ?

**Review by mjsax — COMMENTED** [review]
Thanks @JimGalasyn. Some comments.

File: docs/streams/developer-guide/processor-api.html (hunk 3 above)
Comment (JimGalasyn) [reply docs/streams/developer-guide/processor-api.html]: Moved these paras into the table.

**Comment by mjsax** [comment]
Thanks @JimGalasyn.

Merged to `trunk` and cherry-picked to `2.6` branch.

**Pull request merged**